CMakeLists.txt

```diff
@@ -312,14 +312,14 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 
     elseif(TOOLCHAIN STREQUAL "CLANG")
         SET(CMAKE_C_COMPILER clang)
-        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall  -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -flto -DUSING_LTO -Werror-implicit-function-declaration)
+        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -flto -DUSING_LTO -Werror-implicit-function-declaration)
         if(CMAKE_BUILD_TYPE STREQUAL "Debug")
            ADD_COMPILE_OPTIONS(-g)
         endif()
         if (ARCH STREQUAL "x64")
             ADD_COMPILE_OPTIONS(-mno-red-zone)
         endif()
-        SET(OPENSSL_FLAGS -include base.h -Wno-error=maybe-uninitialized -Wno-error=format -Wno-format -Wno-error=unused-but-set-variable -Wno-cast-qual)
+        SET(OPENSSL_FLAGS -std=c11 -include base.h -Wno-error=format -Wno-format -Wno-cast-qual)
```

Comment:
Why does OpenSSL require C11?

Reply:
https://github.com/DMTF/libspdm/blob/main/os_stub/openssllib/include/openssl/opensslconf.h#L55-#L68
 redefinition of typedef 'INT' is a C11 feature

Reply:
https://github.com/DMTF/libspdm/blob/main/os_stub/openssllib/include/openssl/opensslconf.h#L55-#L68

redefinition of typedef 'INT' is a C11 featur

Reply:
That's probably because `opensslconf.h` doesn't contain `#ifndef ...` include guards. Looks like they fixed it in 2019 : https://github.com/openssl/openssl/pull/9623. @jyao1 How old is this `opensslconf.h` file? Looks like we should generate a new one.

Reply:
Alternatively you can edit the file, even though it says not to, and add proper `#ifndef OPENSSLCONF_H` include guards.